nhacp-specification.md

```diff
@@ -825,6 +827,69 @@ Create a directory at the specified location.
 
 Possible responses: OK, ERROR
 
+### FILE-GETPROP
+
+Get a file descriptor property value.  If the requested property is not
+valid for the file descriptor, the request MUST fail with an EINVAL error.
+
+| Name  | Type | Notes               |
+|-------|------|---------------------|
+| type  | u8   | 0x13                |
+| fdesc | u8   | File descriptor     |
+| which | u16  | The property to get |
+
+Possible responses: UINT32-VALUE, ERROR
+
+The following properties are defined:
+
+| Name    | Value | Notes                                       |
+|---------|-------|---------------------------------------------|
+| DTYPE   | 0x00  | Descriptor type (see below)                 |
+| NBIO    | 0x01  | Non-blocking I/O 1=enabled 0=disabled       |
+| RAVAIL  | 0x02  | Number of bytes available to read           |
```

Comment:
I'm getting rid of RAVAIL and WAVAIL.  There is not a good portable way to implement them (at least on Unix-like platforms).  Instead I will add a FILE-POLL operation that maps roughly to Unix poll(2).